src/llmcompressor/transformers/utils/helpers.py

```diff
@@ -54,3 +59,101 @@ def detect_last_checkpoint(
             )
 
     return last_checkpoint
+
+
+def is_model_quantized_from_path(path: str) -> bool:
+    """
+    Determine if model from path is quantized based
+    on the config
+
+    :param path: path to the model or HF stub
+    :return: True if config contains quantization_config from the given path
+
+    """
+    config = AutoConfig.from_pretrained(path)
+    if config is not None:
+        if hasattr(config, "quantization_config"):
+            return True
+    return False
+
+
+def infer_recipe_from_model_path(model_path: Union[str, Path]) -> Optional[str]:
+    """
+    Infer the recipe from the model_path.
+
+    :param model_path: The path to the model to load. It can be one of the following:
+        - a path to the model directory
+        - a path to the model file
+        - Hugging face model ID
+    :return: The path to the recipe file if found, None otherwise.
+    """
+    model_path = model_path.as_posix() if isinstance(model_path, Path) else model_path
+
+    if os.path.isdir(model_path) or os.path.isfile(model_path):
+        # Model path is a local path to the model directory or file
+        model_path = (
+            os.path.dirname(model_path) if os.path.isfile(model_path) else model_path
+        )
+        recipe = os.path.join(model_path, RECIPE_FILE_NAME)
+
+        if os.path.isfile(recipe):
+            logger.info(f"Found recipe in the model_path: {recipe}")
+            return recipe
+        logger.debug(f"No recipe found in the model_path: {model_path}")
+        return None
+
+    # If the model path is a Hugging Face model ID
+    recipe = recipe_from_huggingface_model_id(model_path)
+
+    if recipe is None:
+        logger.info("Failed to infer the recipe from the model_path")
+
+    return recipe
+
+
+def recipe_from_huggingface_model_id(
+    model_path: str, recipe_file_name: str = RECIPE_FILE_NAME
+) -> Optional[str]:
+    """
+    Attempts to download the recipe from the Hugging Face model ID.
+
+    :param model_path: Assumed to be the Hugging Face model ID.
+    :param recipe_file_name: The name of the recipe file to download.
+     Defaults to RECIPE_FILE_NAME.
+    :return: A tuple:
+        - The path to the recipe file if found, None otherwise.
+        - True if model_path is a valid Hugging Face model ID, False otherwise.
+    """
+    model_id_url = os.path.join(HUGGINGFACE_CO_URL_HOME, model_path)
```

Comment:
can we target just downloading the recipe?

Reply:
yes it does